tests/client/components/carousel.spec.js

```diff
@@ -5,126 +5,434 @@ import { mount } from '@vue/test-utils';
+    it.skip('should have the resistanceCoef set to 20 by default', () => {});
+
+    it.skip('should set the resistanceCoef to 30', () => {});
+  });
+
+  describe('Scrolling per page', () => {
+    it.skip('should have scroll per page set to true by default', () => {});
+
+    // TODO: Reconsider updating this test case name
+    it('should go to second slide when we have odd number of slides and recompute carousel width', done => {
+      const wrapper = mount(Carousel, {
+        propsData: {
+          scrollPerPage: true,
+          perPage: 2
+        },
+        slots: {
+          default: [Slide, Slide, Slide, Slide, Slide]
+        }
+      });
+      wrapper.vm.carouselWidth = 500;
+
+      wrapper.vm.$nextTick(() => {
+        wrapper.vm.goToPage(1);
+        wrapper.vm.computeCarouselWidth();
+
+        expect(wrapper.vm.currentPage).toBe(1);
+
+        done();
+      });
     });
 
-    expect(wrapper.vm.slideCount).toBe(3);
+    it.skip('should advanced the page by only one slide when scroll per page is false', () => {});
+  });
 
-    wrapper.vm.$nextTick(() => {
-      expect(wrapper).toMatchSnapshot();
-      done();
+  describe('Space padding', () => {
+    it.skip('should have 0 space padding by default', () => {});
+
+    it.skip('should have 0 space padding maximum offset factor by default', () => {});
+
+    it.skip('should have 10 space padding', () => {});
+
+    it.skip('should have 10 space padding maximum offset factor by default', () => {});
+  });
+
+  describe('Speed', () => {
+    it.skip('should have a speed of 500 by default', () => {});
+
+    it.skip('should have a speed of 600', () => {});
+  });
+
+  describe('Tag name', () => {
+    it.skip('should only count slides matching the default tagName of slide', () => {});
+
+    it('should only count slides matching tagName', done => {
+      const CustomSlide = {
+        extends: Slide,
+        name: 'CustomSlide',
+      };
+      const wrapper = mount(Carousel, {
+        propsData: {
+          tagName: 'CustomSlide'
+        },
+        slots: {
+          default: [CustomSlide, CustomSlide, Slide]
+        }
+      });
+      expect(wrapper.vm.tagName).toBe('CustomSlide');
+      expect(wrapper.vm.slideCount).toBe(2);
+      wrapper.vm.$nextTick(() => {
+        expect(wrapper).toMatchSnapshot();
+        done();
+      });
     });
   });
 
+  describe('v-model with value prop', () => {
+    it.skip('should set currentPage to 1', () => {});
+
+    it.skip('should not change the current page when selecting the same page', () => {});
+
+    it.skip('should not change the current page when value is negative', () => {});
+
+    it.skip('should not change the current page when value is greater than pageCount', () => {});
+  });
+
+  describe('Events', () => {
+    it.skip('should emit a pageChange event with the page number on page change', () => {});
+
+    it.skip('should emit a slideClick event with the object?? blah on slide click', () => {});
```

Comment:
Not sure what this is supposed to say lol

Reply:
Yeah I was confused about the event but I wanted to be sure to test it :1st_place_medal: